src/widgets/immunizations/immunizations-form.component.tsx

```diff
@@ -0,0 +1,348 @@
+
+    const immunization: Immunization = {
+      patientUuid: patientUuid,
+      immunizationObsUuid: immunizationObsUuid,
+      vaccineName: vaccineName,
+      vaccineUuid: vaccineUuid,
+      manufacturer: manufacturer,
+      expirationDate: vaccinationExpiration,
+      vaccinationDate: vaccinationDate,
+      lotNumber: lotNumber,
+      currentDose: currentDose,
+      sequences: immunizationSequences
+    };
+    const abortController = new AbortController();
+
+    savePatientImmunization(
+      mapToFhirImmunizationResource(
+        immunization,
+        currentVisitUuid,
+        currentLocationUuid,
+        currentProviderUuid
+      ),
+      patientUuid,
+      immunizationObsUuid,
+      abortController
+    ).then(response => {
+      response.status == 200 && navigate();
+    }, createErrorHandler);
+    return () => abortController.abort();
+  };
+
+  function navigate() {
+    history.push(`/patient/${patientUuid}/chart/immunizations`);
+    props.closeComponent();
+  }
+
+  function createForm() {
+    const addFormHeader = t("add vaccine", "Add Vaccine") + ": " + vaccineName;
+    const editFormHeader =
+      t("edit vaccine", "Edit Vaccine") + ": " + vaccineName;
+    return (
+      <form
+        onSubmit={handleFormSubmit}
+        data-testid="immunization-form"
+        onChange={() => {
+          setFormChanged(true);
+          return props.entryStarted();
+        }}
+        className={styles.immunizationsForm}
+        ref={formRef}
+      >
+        <SummaryCard
+          name={viewEditForm ? editFormHeader : addFormHeader}
+          styles={{
+            width: "100%",
+            background: "var(--omrs-color-bg-medium-contrast)",
+            height: "auto"
+          }}
+        >
+          <div className={styles.immunizationsContainerWrapper}>
+            <div style={{ flex: 1, margin: "0rem 0.5rem" }}>
+              {hasSequences(immunizationSequences) && (
+                <div className={styles.immunizationsInputContainer}>
+                  <label htmlFor="sequence">{t("sequence", "Sequence")}</label>
+                  <div className="omrs-select">
+                    <select
+                      id="sequence"
+                      name="sequence"
+                      value={currentDose.sequenceNumber}
+                      onChange={onDoseSelect}
+                      className={`immunizationSequenceSelect`}
+                      required
+                    >
+                      <option value="DEFAULT">
+                        {t("please select", "Please select")}
+                      </option>
+                      {immunizationSequences.map(s => {
+                        return (
+                          <option
+                            key={s.sequenceNumber}
+                            value={s.sequenceNumber}
+                          >
+                            {t(s.sequenceLabel, s.sequenceLabel)}
```

Comment:
Are we sure we can translate `sequenceLabel`? Is this coming from the API (I guess so)?

Reply:
Atleast not right now. Right now it is coming as module config which is configured as a JSON. We had decided to use the JSON configuration as a label than hardcoding it so it could be translated.